### PR TITLE
Fix creation of macvlan interfaces

### PR DESCRIPTION
**Pull request by danwinship**
We need to create the macvlan in the default namespace (so it can refer to the default interface) and then move it.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1389717 (bug 1389717)

@dcbw / @openshift/networking PTAL

### CODE REVIEW

**Comment by knobunc** [comment]
[test]

**Comment by danwinship** [comment]
pushed another commit; we still had macvlan-related code in the shell script even though it never got used any more

**Comment by pravisankar** [comment]
LGTM

**Comment by knobunc** [comment]
[test] flake seems to be the same as https://github.com/openshift/origin/issues/11685

**Comment by danwinship** [comment]
flake #10336, [test]

**Comment by knobunc** [comment]
[merge]

File: pkg/sdn/plugin/pod_linux.go [hunk 1]
```diff
@@ -182,28 +182,36 @@ func addMacvlan(netns string) error {
...
+	if err != nil {
+		return fmt.Errorf("could not open netns %q", netns)
+	}
+	defer ns.Close()
+
+	err = netlink.LinkAdd(&netlink.Macvlan{
+		LinkAttrs: netlink.LinkAttrs{
+			MTU:         defIface.Attrs().MTU,
+			Name:        "macvlan0",
+			ParentIndex: defIface.Attrs().Index,
```
Comment (dcbw) [inline pkg/sdn/plugin/pod_linux.go]: You can give LinkAdd() the netns like so:

```
Namespace:   netlink.NsFd(int(ns.Fd())),
```

and it'll get put into the namespace after it's created.  Then I'd do "ns.Do()" for the LinkByName and LinkSetUp.

**Comment by danwinship** [comment]
@dcbw: updated
@knobunc: this will need a new merge tag after dcbw reviews it, because I deleted your previous one

**Comment by dcbw** [comment]
LGTM [testextended][extended:networking]

**Comment by dcbw** [comment]
@danwinship needs a gofmt on pkg/sdn/plugin/pod_linux.go

**Comment by knobunc** [comment]
[merge]

**Pull request merged**